examples/i3c_target_set_ids.py

```diff
@@ -0,0 +1,130 @@
+from supernovacontroller.sequential import SupernovaDevice
+from BinhoSupernova.commands.definitions import DdrOk
+from BinhoSupernova.commands.definitions import (I3cTargetMemoryLayout_t, I3cTargetMaxDataSpeedLimit_t, I3cTargetIbiCapable_t, I3cTargetIbiPayload_t, 
+                                                 I3cTargetOfflineCap_t, I3cTargetVirtSupport_t, I3cTargetDeviceRole_t, I3cTargetDcr_t, I3cBcrRegister_t,
+                                                 IgnoreTE0TE1Errors, MatchStartStop, AlwaysNack)
+
+def main():
+    """
+    Example to test setting of IDs for the Supernova as an I3C target  with SupernovaController.
+    With two Supernovas connected between them via the HV bus, one is initalized as an I3C controller and the other
+    one as an I3C target. The target acts as a memory with 1024 registers of 1 bytes size
+    
+    Sequence of commands:
+    - Initialize the Devices and create its interfaces: 
+      Create and open a connection to the Supernova host adapter for both the I3C controller and the I3C target.
+      Create an I3C interface for communication with both the I3C controller and the I3C target.
+    - Initialize the I3C peripheral of the target: 
+      Init the peripheral of one Supernova as an I3C target with a memory layout of 1024 registers of 1 byte size.
+    - Set the target PID, BCR, DCR and static address.
+    - Initialize the I3C peripheral of the controller. 
+    - Set I3C Parameters and initialize the Bus: Set transfer rates and initialize the I3C bus with a specific voltage level.
+      For this specific example, the bus initialization assigns the dynamic addresses using an RSTDAA + ENTDAA.
+    - Discover Targets on I3C Bus: Fetch a list of devices present on the I3C bus.
+    - Assign dynamic addresses using RSTDAA and SETDASA.
+    - Close Device Connection: Close the connection to the Supernova devices.
+    """
+
+    print("-----------------------")
+    print("Initialize Supernovas")
+    print("-----------------------")
+
+    target_device = SupernovaDevice()
+    info = target_device.open(usb_address ="\\\\?\\HID#VID_1FC9&PID_82FC#6&bb45e52&1&0000#{4d1e55b2-f16f-11cf-88cb-001111000030}")
```

Comment:
are we cool with hard-coded addresses? If not, let's make tickets to address this in the future.

Reply:
Given that currently we have the `openAllConnectedSupernovaDevices()` method in `SupernovaDevice` class, we should use it across all examples. I just created [this](https://focusuy.atlassian.net/browse/BMC2-1402) ticket to address that. Thanks for the observation!